src/hdf5/cloudsat_netcdf_to_ioda.py

```diff
@@ -95,55 +137,58 @@ def main(args):
-    cpr_obs = cpr_obs.transpose("Location", "channel")
+    file_obs_data = file_obs_data.rename_vars({"elevation": "elevation1"})
+    file_obs_data = file_obs_data.stack(Location=['obs_id', 'elevation']).reset_index("Location")
+    file_obs_data = file_obs_data.transpose("Location", "channel")
 
-    locid = (cpr_obs.obs.values < -100) | (cpr_obs.obs.values > 100) | np.isnan(cpr_obs.obs.values) | np.isinf(cpr_obs.obs.values)
-    locid = cpr_obs.Location.values[np.sum(locid, axis=1) == 0]
-    cpr_obs = cpr_obs.isel(Location=locid)
+    locid = (file_obs_data.obs.values < -100) | (file_obs_data.obs.values > 100) | np.isnan(file_obs_data.obs.values) | np.isinf(file_obs_data.obs.values)
+    locid = file_obs_data.Location.values[np.sum(locid, axis=1) == 0]
+    file_obs_data = file_obs_data.isel(Location=locid)
     # end conditioning and cleansing block
 
     # this function will map the cloud radar data in the cpr xarray
     # into a dictionary to be passed to the IODA writer functions
-    nobs = cpr_obs.Location.size
-    nchans = cpr_obs.channel.size
+    nobs = file_obs_data.Location.size
+    nchans = file_obs_data.channel.size
 
     # ideally an attribute from the file read in
-    # example: cpr_obs.attrs['ShortName'].decode("utf-8")
+    # example: file_obs_data.attrs['ShortName'].decode("utf-8")
     # here we are hardcoding to the function the name of the satellite
-    WMO_sat_ID = get_WMO_satellite_ID('CloudSat')
+    WMO_sat_ID = get_WMO_satellite_ID(sensor_name)
 
     # allocate space for output depending on which variables are to be saved
     obs_data = init_obs_loc()
 
-    obs_data[('latitude', metaDataName)] = cpr_obs.lat.values.astype(np.float32)
-    obs_data[('longitude', metaDataName)] = cpr_obs.lon.values.astype(np.float32)
+    obs_data[('latitude', metaDataName)] = file_obs_data.lat.values.astype(np.float32)
+    obs_data[('longitude', metaDataName)] = file_obs_data.lon.values.astype(np.float32)
     obs_data[('satelliteIdentifier', metaDataName)] = np.full((nobs), WMO_sat_ID, dtype='int32')
-    obs_data[('sensorCentralFrequency', metaDataName)] = np.array([94.05]).astype(np.float32)    # cpr_sim.Frequency.values.astype(np.float32)
-    obs_data[('sensorCentralWavenumber', metaDataName)] = np.array([3.1371]).astype(np.float32)  # cpr_sim.Wavenumber.values.astype(np.float32)
-    obs_data[('sensorPolarizationDirection', metaDataName)] = np.array([9]).astype(np.int32)
-    obs_data[('sensorScanPosition', metaDataName)] = cpr_obs.fov1.values.astype(np.int32)
-    obs_data[('sensorChannelNumber', metaDataName)] = cpr_obs.channel.values.astype(np.int32)
+    obs_data[('sensorCentralFrequency', metaDataName)] = file_obs_data.centerFreq.values.astype(np.float32)
+    obs_data[('sensorCentralWavenumber', metaDataName)] = file_obs_data.centerWN.values.astype(np.float32)
+    obs_data[('sensorPolarizationDirection', metaDataName)] = np.full((nobs), 9, dtype='int32')
```

Comment:
@imoradi what is this value?  Is it in the provided files, it appears it's set to a hardcoded value of `9` for both cloudsat and GPM DPR?

@BenjaminTJohnson is this needed?